infrastructure/helm/quantumserverless/README.md

```diff
@@ -109,6 +115,26 @@ For our Ray Charts dependencies we are using the configuration created by the Ra
 
 - For Ray Api Server you can read their [values.yaml](https://github.com/ray-project/kuberay-helm/blob/main/helm-chart/kuberay-apiserver/values.yaml).
 
+- Ray Api Server access needs the access token issued by the keycloak.  Here is the example to obtain the access token and send request to the Ray API Server
+
+```
+#!/bin/bash
+API=$1
+RESPONSE=$(curl --request POST \
+  --url 'http://<LOCAL-IP>:31059/realms/quantumserverless/protocol/openid-connect/token' \
+  --header 'content-type: application/x-www-form-urlencoded' \
+  --data grant_type=client_credentials \
+  --data client_id=rayapiserver \
+  --data client_secret=APISERVERSECRET-CHANGEME \
```

Comment:
I got an `{"error":"unauthorized_client","error_description":"Invalid client or Invalid client credentials"}` error when running this command. I've double checked that the client_secret value is correct. Good chance this is a continuation of the issues I was having yesterday with keycloak...

Reply:
you can probably see the reason in the log of the apiserver pod.(gatekeeper thread).

Reply:
nothing in the gatekeeper logs, but did find this in the keycloak ones:

```console
$ k logs -n ray quantum-serverless-keycloak-0 | grep ERROR
2023-03-03 20:03:45,616 WARN  [org.keycloak.events] (executor-thread-4) type=CLIENT_LOGIN_ERROR, realmId=e165160a-1516-4a29-b65c-6f4479682dc3, clientId=rayapiserver, userId=null, ipAddress=10.244.0.1, error=client_not_found, grant_type=client_credentials
2023-03-03 20:04:09,874 WARN  [org.keycloak.events] (executor-thread-4) type=CLIENT_LOGIN_ERROR, realmId=e165160a-1516-4a29-b65c-6f4479682dc3, clientId=rayclient, userId=null, ipAddress=10.244.0.1, error=invalid_client_credentials, grant_type=client_credentials
```

worth noting that the client ID in the bash command is `rayapiserver`, but when trying that I got a `client not found` error. So I tried using `rayclient` (which is what I saw in the UI), but that failed for a different reason - `invalid client credentials`.

Reply:
@psschwei Thanks for trying.  I tried myself with kind on an ubuntu.  It worked fine.  I specified the **172.18.0.2** as the local-ip.  From your error messages, the keycloak configuration file might not be read by the Keycloak pod.  It is read only on the start up of the pod.  Would you check if the `rayapiserver` client is defined in the `quntumserverless` realm in the keycloak console?

Thanks!

Reply:
ok, it's working on a fresh setup this morning